cfme/storage/volume.py

```diff
@@ -243,6 +287,36 @@ def create_snapshot(self, name, cancel=False, reset=False):
                 view.save.click()
                 return snapshot_collection.instantiate(name, self.provider)
 
+    def attach_instance(self, name, mountpoint=None, cancel=False, reset=False):
+        view = navigate_to(self, 'AttachInstance')
+
+        # Reset and Attach buttons are only active when view is changed
+        changed = view.fill({'instance': name, 'mountpoint': mountpoint})
+
+        if cancel:
```

Comment:
I think this is close, but we are missing a condition. What is the form was not changed? What do you think about updating the cancel statement to:

```
if cancel or not was_changed:
    if not was_changed:
        logger.info("attach_instance: the form was unchanged")
    self.click.cancel()
```

Reply:
Right, I agree